src/ess/reflectometry/corrections.py

```diff
@@ -0,0 +1,105 @@
+    velocity.unit = sc.units.m / sc.units.s
+    y_measured = sc.geometry.y(pixel_position)
+    z_measured = sc.geometry.z(pixel_position)
+    z_origin = sc.geometry.z(sample_position)
+    y_origin = sc.geometry.y(sample_position)
+    y_dash = y_dash0(velocity, z_origin, y_origin, z_measured, y_measured)
+    intercept = y_origin - y_dash * z_origin
+    y_true = z_measured * y_dash + intercept
+    angle = sc.Variable(
+        values=np.degrees(
+            sc.atan(y_true / z_measured).bins.constituents["data"].values),
+        unit=sc.units.deg,
+        dims=["event"],
+    )
+    return angle
+
+
+def y_dash0(velocity, z_origin, y_origin, z_measured, y_measured):
+    """
+    Evaluation of the first dervative of the kinematic equations for for the trajectory of a neutron reflected from a surface.
+
+    Args:
+        velocity (`scipp._scipp.core.VariableView`): Neutron velocity.
+        z_origin (`scipp._scipp.core.Variable`): The z-origin position for the reflected neutron.
+        y_origin (`scipp._scipp.core.Variable`): The y-origin position for the reflected neutron.
+        z_origin (`scipp._scipp.core.Variable`): The z-measured position for the reflected neutron.
+        y_origin (`scipp._scipp.core.Variable`): The y-measured position for the reflected neutron.
+
+    Returns:
+        (`scipp._scipp.core.VariableView` or `array_like`): The gradient of the trajectory of the neutron at the origin position.
+    """
+    return ((-G_ACC * (z_measured - z_origin) * (z_measured - z_origin) /
+             (2 * velocity * velocity)) - y_origin +
+            y_measured) / (z_measured - z_origin)
+
+
+def illumination_correction(beam_size, sample_size, theta):
+    """
+    The factor by which the intensity should be multiplied to account for the
+    scattering geometry, where the beam is Gaussian in shape.
+
+    Args:
+        beam_size (:py:attr:`sc.Variable`): Width of incident beam.
+        sample_size (:py:attr:`sc.Variable`): Width of sample in the dimension of the beam.
+        theta (:py:attr:`sc.Variable`): Incident angle.
+
+    Returns:
+        (:py:attr:`array_like`): Correction factor.
+    """
+    beam_on_sample = beam_size / sc.sin(theta)
+    scale_factor = erf((sample_size / beam_on_sample * 2.35482).values)
+    return scale_factor
+
+
+def illumination_of_sample(beam_size, sample_size, theta):
+    """
+    Determine the illumination of the sample by the beam and therefore the size of this illuminated length.
+
+    Args:
+        beam_size (:py:attr:`sc.Variable`): Width of incident beam, in metres.
+        sample_size (:py:attr:`sc.Variable`): Width of sample in the dimension of the beam, in metres.
+        theta (:py:attr:`sc.Variable`): Incident angle.
+
+    Returns:
+        (`sc.Variable`): The size of the beam, for each theta, on the sample.
+    """
+    beam_on_sample = beam_size / sc.sin(theta)
+    if ((sc.mean(beam_on_sample)) > sample_size).values:
```

Comment:
```suggestion
    if ((sc.mean(beam_on_sample)) > sample_size).value:
```
if this is expected to have only a single value?